Travis OSX Builds

RowenStipe opened the pull request:
This pull requests adds in files and code to implement Travis to create a distributable `qTox.dmg` file with an image background containing:
- README.md (From main qTox dir)
- LICENCE (From main qTox dir)
- Symbolic link to `\Applications\` named 'Link to Applications'
- A use-able `qTox.app`

The `.travis.yml` does require further editing to add in the the API key and un-comment lines to actually implement this new system but everything else is in place to do so.

Also an extra development variable was added to `./osx/qTox-Mac-Deployer-ULTIMATE.sh` because of work flow. And just because. Plus the command section rewritten to be cleaner.

SkyzohKey commented on osx/qTox-Mac-Deployer-ULTIMATE.sh:
```diff
@@ -245,52 +248,64 @@ function bootstrap() {
-if [[ "$1" == "-i" ]]; then
+function dmgmake() {
+	fcho "------------------------------"
+	fcho "Starting DMG creation"
```
Shouldn't `fcho` be `echo` ?

RowenStipe replied:
Said this in IRC but for context `fcho` is a custom function used in various points throughout the script. https://github.com/tux3/qTox/blob/master/osx/qTox-Mac-Deployer-ULTIMATE.sh#L53

RowenStipe committed:
chore(travis): Allow for Travis deployment

Allows for OSX packages through travis, .travis.yml requires editing

chore(travis): Let Travis deploy OSX

Allows Travis to start Deploying when ever a tag is applied.

zetok commented on .travis.yml:
```diff
@@ -1,14 +1,25 @@
+# deploy:
+#   provider: releases
+#   api_key:
+#     secure: Ji05dTgtKeOirakH6Y4N79Zm6eAeD3rAv0xNqis/mljSi6PxsVn2CqVJyqpHhy4yw6vutjFgxEBbLsaNFv6TcPli3lj1HsrUssYGsKPz0VTl4kxRPCiP+NjsBiuU4/tzb8kSyDkIA/spPwwmEWCEID8BtIDESIcQWMpInfqfMh/yxcK31e2X5AxWbWb90D9smS5B+Yee16KiDCzmc81vHO6k6TpSFxQifrUp5iS6xCLNxqGs6LUv6zBHVLmc2FxT4uNqXQewfV0fg7oW0ghTFo/k8CzfzPx0PHFzZcdmwDFXOX06cbCe1SAnvJnvrBFSnF2qrDMmATMZjGQlDEvWrWhEOZwYV4ntFcvnVTpuI+S+Q6QjREeJkwos2lAJ4bCPJ ...
```
IMO this section should be uncommented, this value removed, and replaced with something like
`<fill it to make it work>`

RowenStipe replied:
I never tested to see what might happen if this section was un-commented but I just didn't want to risk possibly breaking things by doing so. The one change I might do though is to remove the trailing spaces that exist after `#` since travis is as picky about spaces as Python can be so only the `#` needs to be removed.

zetok replied:
Yeah, it would remain broken until @tux3 would decide to "fix" it.

tux3 replied:
I added a deployment key under the `$GH_TOKEN` environment variable, and I created an encrypted oauth token:
`secure: "BRbzTWRvadALRQSTihMKruOj64ydxusMUS9FQR//qFlS345ZYfYta43W//4LcWWDKtj6IvA6DRqNdabgWnpbpxpnm9gVftGUdOKlU3niPZhwsMkB2M12QHUnAP6DVOfGPvdciBV+6mu73SSxniEcrYjZ1CrRX7mknmehPpVKxNk="`

ddobrev commented:
@zetok @tux3 @RowenStipe please fix that uncommented section and merge this pull, my friends and I have been waiting for OS X builds for months.

RowenStipe commented:
With my last commit, Travis is now ready to deploy when ever a tag is applied to any branch on the repo with this commit in it.
